frontend/packages/ux-editor/src/components/TextResourceEdit.tsx

```diff
@@ -69,13 +69,19 @@ const TextBox = ({ language, t, textResource, textResourceId }: TextBoxProps) =>
   const { org, app } = useParams();
   const { mutate } = useUpsertTextResourcesMutation(org, app);
 
-  const updateTextResource = (text: string) =>
-    mutate({ language, textResources: [{ id: textResourceId, value: text, variables: textResource.variables }] });
+  const updateTextResource = (text: string) => {
+    if (textResource) {
```

Comment:
Another problem I now see with this extra check is that we are just hiding the error. If the problem happens again, we won't be able to catch the error and display a generic error message to the user

Reply:
no more extra check , I removed it :slightly_smiling_face:, I am not sure if we need to display like this error?

Reply:
Yes, but I see that you still use an optional chaining operator 😉
`textResources: [{ id: textResourceId, value: text, variables:textResource?.variables }],` 

As I understand it, the textResource should never be null or undefined when we update the resource. So I don't think that we need to handle it since it's not an expected behavior.
I think that we should display a generic error message and even log the error.

Reply:
I tested it with multiple languages and I get the runtime error too, so I was wrong, we need the optional chaining operator 👍